src/main/java/mate/dao/DriverDaoImpl.java

```diff
@@ -70,6 +70,18 @@ public List<Driver> getAll() {
         }
     }
 
+    @Override
+    public Driver getByLicenceNumber(String licenceNumber) {
+        Driver driver = getAll()
+                .stream()
+                .filter(f -> f.getLicenseNumber().equals(licenceNumber))
+                .findFirst()
+                .orElseThrow(
+                        () -> new DataProcessingException("can not find driver by licence number"
+                                + licenceNumber));
+        return driver;
+    }
+
```

Comment:
if you're not using the queries and don't access db here, we can move such methods to the service layer